### ISSUE TITLE
Cloudmanager provider issue - TFE plan try to create an 4 month existing CM, like if not existing...

### ISSUE THREAD

**Issue by philippecotter** [NetApp/terraform-provider-netapp-cloudmanager#45]
Hello.

Since yesterday afternoon, we are no more able to provide any resources (AWS EFS, FsX, S3 buckets,... ) in
our 4 PROD accounts in AWS via Terraform, because when running a simple terraform plan
for any other instance, EFS or else, not
related to our CMO's deployment done via the NetApp provider, the pipeline try to create a new cloud manager like if it do not find
the existing one (existing for
more than 4 months) .

I verified the workspace state, it is listed in there, and no changes were
done since the last apply.

Precision: we use Terraform Enterprise v.0.13.5 and the netapp-cloudmanager Provider of Hashicorp
registry, version 21.1.1

-----------------------------------------
Bellow you see the log of terraform
symptoms,

Terraform v0.13.5
Configuring remote state backend...
Initializing Terraform configuration...
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

....................

------------------------------------------------------------------------
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create
  ~ update in-place
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # netapp-cloudmanager_connector_aws.xxxxxxxxxxxxxxxxxx **will be created**
  + resource "netapp-cloudmanager_connector_aws" "xxxxxxxxxxxxx" {
      + account_id                    = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
      + associate_public_ip_address   = false
      + client_id                     = (known after apply)
      + company                       = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
      + enable_termination_protection = false
      + iam_instance_profile_name     = "occm"
      + id                            = (known after apply)
      + instance_type                 = "t3.xlarge"
      + key_name                      = "NetApp"
      + name                          = "ncmopgr01"
      + proxy_certificates            = [
          + "./xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.crt",
        ]
      + proxy_url                     = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
      + region                        = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
      + security_group_id             = "xxxxxxxxxxxxxxxxxxxxxx"
      + subnet_id                     = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
.
.
.

     
    }

Plan: **1 to add**, 0 to change, 0 to destroy.
```

**Comment by lonico** [comment]
Customer reported that this is most likely due to a change in AMI id for CloudManager in AWS marketplace.

They were able to fix the issue by specifying the AMI id in the connector TF config file, and inserting the older AMI id in the state file.
We'll need to understand better, but I would think that we should have been able to query all existing OCCM instances, regardless of the AMI id if it is not specified.   On the other hand, it seems we should have recorded the AMI id in the state file.

**Comment by lonico** [comment]
Also, "by design", our current provider does not support update operations for connector and CVO.  We'll need to see how this could have contributed to the issue, and provide more information when an update should be appropriate.  And start supporting update operations where useful.

**Comment by lonico** [comment]
One issue with resourceOCCMAWSExists id that if the AMI is not present, we query a list of AMIs for OCCM, and use the latest.  While the customer instance may use an older value.

**Comment by lonico** [comment]
fixed in a46401d67704dc50cddb597beff92263fe59d1da

released as 21.6.0
